### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by akshay1217**
Login & Signup Feature.

### CODE REVIEW

File: app/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        password,
+        
+    });
+
+    // Save user data in the database
+
+    User.find({email : req.body.email.toString().toLowerCase()})
+                .then(user => {
+                    if (user.length > 0){
+                        res.send({errorMessage: "User already exist, try another Email ID."})
```
Comment (bhoomika911) [inline app/controllers/user.controller.js:30]: Add all this messages in translation file.

File: app/controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                // console.log(user)
+                var token = jwt.sign({ userId: user[0]._id}, config.privateKey);
+
+                res.send({
+                    message: "Login Successful!",
+                    userId : user[0]._id,
+                    token
+                })
+            }else {
+                res.send({errorMessage: "Incorrect credentials.! Login Failed."});
```
Comment (bhoomika911) [inline app/controllers/user.controller.js:68]: Do the same for all hard coded message

File: index.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+// create express app
+const app = express();
+
+// parse requests of content-type - application/x-www-form-urlencoded
+app.use(bodyParser.urlencoded({ extended: true }))
+
+// parse requests of content-type - application/json
+app.use(bodyParser.json())
+
+app.use(function (req, res, next) {
```
Comment (bhoomika911) [inline index.js:13]: Why we need this function?

File: app/controllers/user.controller.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
+const User = require('../models/user.model.js');
+const jwt = require('jsonwebtoken');
+const config = require('../../config/config.js');
+
+// Create and Save a new user
+exports.signUp = (req, res) => {
+
+    // Validate request
+    if (!req.body.userName || !req.body.email || !req.body.password) {
+
```
Comment (bhoomika911) [inline app/controllers/user.controller.js:10]: Remove unnecessary space.

**Review by bhoomika911 — CHANGES_REQUESTED** [review]
Please check comments!

File: app/controllers/user.controller.js (hunk 4 above)
Comment (akshay1217) [reply app/controllers/user.controller.js:10]: done.

File: index.js (hunk 3 above)
Comment (akshay1217) [reply index.js:13]: This is  defined for Cross origin resource sharing(CORS) for access control.

File: app/controllers/user.controller.js (hunk 1 above)
Comment (akshay1217) [reply app/controllers/user.controller.js]: done.

File: app/controllers/user.controller.js (hunk 2 above)
Comment (akshay1217) [reply app/controllers/user.controller.js]: done.

**Pull request merged**